peer.go

```diff
@@ -0,0 +1,1087 @@
+package raft
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+
+	"github.com/armon/go-metrics"
+)
+
+type peerOptions struct {
+	// Where to print debug messages, or nil for stderr.
+	logger *log.Logger
```

Comment:
This might be better passed in at start time and saved in `peerShared`. We end up with two references as it stands now.